File: app/src/main/java/noahzu/github/io/gank/Common/PhotoActivity.java
package noahzu.github.io.gank.Common;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import noahzu.github.io.gank.R;
import uk.co.senab.photoview.PhotoViewAttacher;

public class PhotoActivity extends AppCompatActivity {
    public static final String IMAGE_URL = "image_url";

    private Toolbar toolbar;
    private String imageUrl;
    private ImageView imageView;
    private ImageView saveImage;
    private ImageView shareImage;
    private PhotoViewAttacher mAttacher;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_photo);

        imageUrl = getIntent().getStringExtra(IMAGE_URL);
        initView();
        initToolbar();
        loadImage();
    }

    private void initView() {
        imageView = (ImageView) findViewById(R.id.beauty_img);
        saveImage = (ImageView) findViewById(R.id.save_img);
        shareImage = (ImageView) findViewById(R.id.share_img);

    }

    private void loadImage() {
        Picasso.with(this).load(imageUrl).into(imageView);
        mAttacher = new PhotoViewAttacher(imageView);
    }

    private void initToolbar() {
        toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);//显示箭头
        toolbar.setTitle("关于");
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {//箭头的点击事件
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: app/src/main/java/noahzu/github/io/gank/Common/PhotoActivity.kt
package noahzu.github.io.gank.Common

import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import com.squareup.picasso.Picasso
import noahzu.github.io.gank.R

/**
 * Created by zujinhao on 17/5/19.
 */
class PhotoActivity : AppCompatActivity() {

    private val IMAGEURL : String = "image_url"
    private lateinit var toolBar : Toolbar
    private lateinit var imageUrl : String
    private lateinit var imageView : ImageView

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        setContentView(R.layout.activity_photo)
        imageUrl = intent.getStringExtra(IMAGEURL)

        initView()
        initToolbar()
        loadImage()
    }
    private fun initView() {
        imageView = findViewById(R.id.beauty_img) as ImageView
    }
    private fun initToolbar() {
        toolBar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolBar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        toolBar.setTitle("关于")
        toolBar.setNavigationOnClickListener { finish() }
    }

    private fun loadImage() {
        Picasso.with(this).load(imageUrl).into(imageView)
    }
}